src/foam/nanos/dig/SUGAR.js

```diff
@@ -231,27 +255,70 @@ foam.CLASS({
           query = true;
 
           if ( argumentInfo[k].value != "" ) {
-            url += query ? "&" : "?";
-            url += argumentInfo[k].name + "=" + argumentInfo[k].value;
+            this.postData += query ? "&" : "?";
+            this.postData += argumentInfo[k].name + "=" + argumentInfo[k].value;
           }
 
           if ( argumentInfo[k].objectType ) {
             var javaType_ = argumentInfo[k].javaType;
             var prop = foam.lookup(javaType_).getAxiomsByClass(foam.core.Property);
-            url += "&" + argumentInfo[k].name + "=" + argumentInfo[k].name + "&";
+            this.postData += "&" + argumentInfo[k].name + "=" + argumentInfo[k].name + "&";
 
             for ( var i = 0 ; i < prop.length ; i++ ) {
               if ( argumentInfo[k].objectType.instance_[prop[i].name] ) {
-                url += query ? "&" : "?";
+                this.postData += query ? "&" : "?";
                 query = true;
 
-                url += prop[i].name + "=" + argumentInfo[k].objectType.instance_[prop[i].name];
+                this.postData += prop[i].name + "=" + argumentInfo[k].objectType.instance_[prop[i].name];
               }
             }
           }
         }
+        if ( (url.length + this.postData.length) >= this.MAX_URL_SIZE ) {
+          this.postURL = url;
+        }
+        return encodeURI(url + this.postData);
+      }
+    },
+    {
+      class: 'String',
+      name: 'result',
+      value: 'No Request Sent Yet.',
+      view: { class: 'foam.u2.tag.TextArea', rows: 5, cols: 137 },
+      visibility: 'RO'
+    }
+  ],
 
-        return encodeURI(url);
+  actions: [
+    {
+      name: 'postButton',
+      hidden: true,
+      label: 'Send POST Request',
+      code: async function() {
+        if ( ! (this.postURL === '') ) {
+          var req = this.HTTPRequest.create({
+            url: this.appConfig.URL.value + this.postURL.substring(1),
+            method: 'POST',
+            contentType: 'url',
+            payload: this.postData.substring(1),
+          }).send();
```

Comment:
Doesn't have to happen in this PR (can be future ticket), but we need to add http authentication info with the sessionid so these posts can proceed with an additional login.